src/Core/Geometry/TopologicalMesh.inl

```diff
@@ -278,6 +280,264 @@ inline void TopologicalMesh::interpolateAllPropsOnFaces(
     interpolatePropsOnFaces( fh, getVector4PropsHandles(), vec4Props );
 }
 
+inline std::set<TopologicalMesh::WedgeIndex>
+TopologicalMesh::getVertexWedges( OpenMesh::VertexHandle vh ) const {
+    std::set<TopologicalMesh::WedgeIndex> ret;
+
+    for ( ConstVertexIHalfedgeIter vh_it = cvih_iter( vh ); vh_it.is_valid(); ++vh_it )
+    {
+        auto widx = property( m_wedgeIndexPph, *vh_it );
+        if ( widx.isValid() && !m_wedges.getWedge( widx ).isDeleted() ) ret.insert( widx );
+    }
+    return ret;
+}
+
+inline const TopologicalMesh::WedgeData&
+TopologicalMesh::getWedgeData( const WedgeIndex& idx ) const {
+    return m_wedges.getWedgeData( idx );
+}
+
+inline void TopologicalMesh::setWedgeData( TopologicalMesh::WedgeIndex widx,
+                                           const TopologicalMesh::WedgeData& wedge ) {
+    m_wedges.setWedgeData( widx, wedge );
+}
+
+template <typename T>
+inline bool TopologicalMesh::setWedgeData( const TopologicalMesh::WedgeIndex& idx,
+                                           const std::string& name,
+                                           const T& value ) {
+    return m_wedges.setWedgeData( idx, name, value );
+}
+
+inline const std::vector<std::string>& TopologicalMesh::getVec4AttribNames() const {
+    return m_wedges.m_vector4AttribNames;
+}
+inline const std::vector<std::string>& TopologicalMesh::getVec3AttribNames() const {
+    return m_wedges.m_vector3AttribNames;
+}
+inline const std::vector<std::string>& TopologicalMesh::getVec2AttribNames() const {
+    return m_wedges.m_vector2AttribNames;
+}
+inline const std::vector<std::string>& TopologicalMesh::getFloatAttribNames() const {
+    return m_wedges.m_floatAttribNames;
+}
+
+inline bool TopologicalMesh::isFeatureVertex( const VertexHandle& vh ) const {
+    return getVertexWedges( vh ).size() != 1;
+}
+
+inline bool TopologicalMesh::isFeatureEdge( const EdgeHandle& eh ) const {
+
+    auto heh0 = halfedge_handle( eh, 0 );
+    auto heh1 = halfedge_handle( eh, 1 );
+
+    return property( m_wedgeIndexPph, heh0 ) ==
+               property( m_wedgeIndexPph,
+                         prev_halfedge_handle( opposite_halfedge_handle( heh0 ) ) ) ||
+           property( m_wedgeIndexPph, heh1 ) ==
+               property( m_wedgeIndexPph,
+                         prev_halfedge_handle( opposite_halfedge_handle( heh1 ) ) );
```

Comment:
according to the doc, an edge is feature if any of the vertices has different wedges on edge's sides.
thus it is not if both have the same on each side, thus:
```suggestion
    return property( m_wedgeIndexPph, heh0 ) !=
               property( m_wedgeIndexPph, prev_halfedge_handle( heh1 ) ) ||
           property( m_wedgeIndexPph, heh1 ) !=
               property( m_wedgeIndexPph, prev_halfedge_handle( heh0 ) );
```

Reply:
hmm why keeping the parts `prev_halfedge_handle( heh* )` since `heh0` and `heh1` are opposite?